armi/reactor/blocks.py

```diff
@@ -1806,7 +1806,9 @@ def setPinPowers(
             self.p.pinPowersNeutron = self.p.pinPowers
 
         if gamma:
-            self.p.pinPowers = self.p.pinPowersNeutron + self.p.pinPowersGamma
+            self.p.pinPowers = [
```

Comment:
I just checked out the comments in physics\neutronics\parameters.py, which suggest that `pinPowers` should be a 1D array but `pinPowersNeutron` and `pinPowersGamma` are supposed to be 2D arrays (numPins x numAxialSegments). Maybe some of the comments are outdated? `setPinPowers` doesn't have any sense of axial segments within the block. 

Since this function essentially is the setter, we could just cast them as numpy arrays here. Is that robust enough, or do we need to prevent other functions being created in the future that could set these variables as a different type?

Reply:
Comments could be out of date. I think T/H is what would  be most affected by this, so that'd be the place to look to get to the bottom of it. It's precisely questions like this that has us pulling lots of these `Block` methods off and closer to the physics interface/plugin code that knows what to do with them; this is a prime candidate for that treatment (should probably be in the neutronics plugin if I had to guess). That should be a separate PR for later.

As for enforcing type, while it appears that this is the only way that pin powers should be set, there is no guarantee that that is the case, and I wouldn't be surprised if there is code out there that puts powers straight onto the parameters. Safest thing is to define a setter on the parameter itself, since one would need to pretty deliberately circumvent it. That would probably be done at the same time as pulling the `setPinPowers()` method off.

Reply:
Turns out this actually affected performance quite a bit, so I might actually implement the param setter convert-to-numpy approach to see if that helps some